### PR TITLE
8278583: Open source SwingMark - Swing performance benchmark

### PR DESCRIPTION
**Pull request by prrace**
This is the SwingMark performance benchmark.
It is very simplistic by today's standards but has been in use since 1998
and gives a sense of changes that affect the UI stack not just core Java 2D
tight loop rendering.
It is mostly "as is" with these exceptions

1) I did some tidying up of the woeful code formatting
2) I migrated some deprecated API and suppressed warnings for others (some future work there, but I didn't want to make changes that might affect comparisons)
3) I added a Makefile and support for building as a jar.
4) TestList.txt is a resource in the .jar so most folks don't need to worry about it
5) README is cleaned up and updated
6) I used the same licensing as J2DBench
7) I've put it alongside RenderPerfTest in the sources

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8278583](https://bugs.openjdk.org/browse/JDK-8278583): Open source SwingMark - Swing performance benchmark

### Reviewers
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**) ⚠️ Review applies to [1841c85e](https://git.openjdk.org/jdk/pull/13486/files/1841c85e0077e3f19a47cb7d4020cfda80bea044)
 * [Ajit Ghaisas](https://openjdk.org/census#aghaisas) (@aghaisas - **Reviewer**) ⚠️ Review applies to [1841c85e](https://git.openjdk.org/jdk/pull/13486/files/1841c85e0077e3f19a47cb7d4020cfda80bea044)
 * [Alexey Ushakov](https://openjdk.org/census#avu) (@avu - Committer) ⚠️ Review applies to [1841c85e](https://git.openjdk.org/jdk/pull/13486/files/1841c85e0077e3f19a47cb7d4020cfda80bea044)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13486/head:pull/13486` \
`$ git checkout pull/13486`

Update a local copy of the PR: \
`$ git checkout pull/13486` \
`$ git pull https://git.openjdk.org/jdk.git pull/13486/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13486`

View PR using the GUI difftool: \
`$ git pr show -t 13486`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13486.diff">https://git.openjdk.org/jdk/pull/13486.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13486#issuecomment-1509589118)

### CODE REVIEW

**Review by mrserb — APPROVED** [review]

**Review by aghaisas — APPROVED** [review]

File: test/jdk/performance/client/SwingMark/src/LabelTest.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+   }
+}
+
+class LabelChanger implements Runnable {
+   JLabel[] labels;
+   Color color;
+
+
+   public LabelChanger(JLabel[] labelsToChange) {
+      labels =  labelsToChange;
```
Comment (turbanoff) [inline test/jdk/performance/client/SwingMark/src/LabelTest.java:114]: nit
```suggestion
      labels = labelsToChange;
```

File: test/jdk/performance/client/SwingMark/src/TableRowTest.java [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+
+class TableScroller implements Runnable {
+    JTable table;
+    int scrollAmount = 1;
+    int currentRowSelection = 0;
+
+
+    public TableScroller(JTable tableToScroll, int scrollBy, int selectionMode) {
+        table =  tableToScroll;
```
Comment (turbanoff) [inline test/jdk/performance/client/SwingMark/src/TableRowTest.java:166]: nit
```suggestion
        table = tableToScroll;
```

File: test/jdk/performance/client/SwingMark/src/TableRowTest.java [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+    JTable table;
+    int currentRowSelection = 0;
+    Vector dataVector;
+    int index = 0;
+    Object [][] data;
+    boolean add = true; // false for "remove"
+
+
+    public TableRowAdder(JTable table, Object[][] data, boolean add) {
+        this.table =  table;
```
Comment (turbanoff) [inline test/jdk/performance/client/SwingMark/src/TableRowTest.java:213]: nit
```suggestion
        this.table = table;
```

File: test/jdk/performance/client/SwingMark/src/TableColTest.java [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+}
+
+
+class TableColScroller implements Runnable {
+    JTable table;
+    int scrollAmount = 1;
+    int currentColSelection = 0;
+
+    public TableColScroller(JTable tableToScroll, int scrollBy, int selectionMode) {
+      table =  tableToScroll;
```
Comment (turbanoff) [inline test/jdk/performance/client/SwingMark/src/TableColTest.java:139]: nit
```suggestion
      table = tableToScroll;
```

File: test/jdk/performance/client/SwingMark/src/ListTest.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+public class ListTest extends AbstractSwingTest {
+    JList list1;
+    final int list1ItemCount = 250;
+    String DISPLAY_STRING = "List Item ";
+
+    public JComponent getTestComponent() {
+        loadBundle();
+        JPanel panel = new JPanel();
+        String[] list1Data =  new String[list1ItemCount];
```
Comment (turbanoff) [inline test/jdk/performance/client/SwingMark/src/ListTest.java:54]: nit
```suggestion
        String[] list1Data = new String[list1ItemCount];
```

File: test/jdk/performance/client/SwingMark/src/ListTest.java [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+}
+
+
+class ListScroller implements Runnable {
+   JList list;
+   int scrollAmount = 1;
+
+
+   public ListScroller(JList listToScroll, int scrollBy) {
+      list =  listToScroll;
```
Comment (turbanoff) [inline test/jdk/performance/client/SwingMark/src/ListTest.java:116]: nit
```suggestion
      list = listToScroll;
```

File: test/jdk/performance/client/SwingMark/src/NullRunnable.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/** This is a simple class to be used as a "do nothing"
+  * it is part of the timing system.
+  *
+  * @see SwingMarkPanel#runTests
+  */
+
+class NullRunnable implements Runnable  {
```
Comment (turbanoff) [inline test/jdk/performance/client/SwingMark/src/NullRunnable.java:38]: nit
```suggestion
class NullRunnable implements Runnable {
```

File: test/jdk/performance/client/SwingMark/src/SliderTest.java [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+            }
+        }
+    }
+
+class SliderInc implements Runnable {
+    JSlider slider;
+    int incAmount = 1;
+
+    public SliderInc(JSlider sliderToIncrement, int incrementBy) {
+        slider =  sliderToIncrement;
```
Comment (turbanoff) [inline test/jdk/performance/client/SwingMark/src/SliderTest.java:103]: nit
```suggestion
        slider = sliderToIncrement;
```

File: test/jdk/performance/client/SwingMark/src/AdvancedTextAreaTest.java [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+        runStandAloneTest(new AdvancedTextAreaTest());
+    }
+
+    static class TextAppender implements Runnable {
+        JTextArea area;
+        String appendString;
+
+
+        public TextAppender(JTextArea textArea, String appendThis) {
+            area =  textArea;
```
Comment (turbanoff) [inline test/jdk/performance/client/SwingMark/src/AdvancedTextAreaTest.java:128]: nit
```suggestion
            area = textArea;
```

File: test/jdk/performance/client/SwingMark/src/AdvancedTextAreaTest.java [hunk 10]
```diff
@@ -0,0 +1,244 @@
...
+        public void run() {
+            area.append(appendString);
+        }
+    }
+
+    static class TextCutter implements Runnable {
+        JTextArea area;
+
+        public TextCutter(JTextArea textArea) {
+            area =  textArea;
```
Comment (turbanoff) [inline test/jdk/performance/client/SwingMark/src/AdvancedTextAreaTest.java:141]: nit
```suggestion
            area = textArea;
```

File: test/jdk/performance/client/SwingMark/src/AdvancedTextAreaTest.java [hunk 11]
```diff
@@ -0,0 +1,244 @@
...
+        public void run() {
+            area.cut();
+        }
+    }
+
+    static class TextCopier implements Runnable {
+        JTextArea area;
+
+        public TextCopier(JTextArea textArea) {
+                area =  textArea;
```
Comment (turbanoff) [inline test/jdk/performance/client/SwingMark/src/AdvancedTextAreaTest.java:153]: nit
```suggestion
            area = textArea;
```

File: test/jdk/performance/client/SwingMark/src/AdvancedTextAreaTest.java [hunk 12]
```diff
@@ -0,0 +1,244 @@
...
+
+    static class TextFontSetter implements Runnable {
+        JTextArea area;
+        String[] fonts;
+        Random random;
+        int index = 0;
+
+        @SuppressWarnings("deprecation")
+        public TextFontSetter(JTextArea textArea) {
+            area =  textArea;
```
Comment (turbanoff) [inline test/jdk/performance/client/SwingMark/src/AdvancedTextAreaTest.java:169]: nit
```suggestion
            area = textArea;
```

File: test/jdk/performance/client/SwingMark/src/AdvancedTextAreaTest.java [hunk 13]
```diff
@@ -0,0 +1,244 @@
...
+            index ++ ;
+            index = index % fonts.length ;
+        }
+    }
+
+    static class TextPaster implements Runnable {
+        JTextArea area;
+
+        public TextPaster(JTextArea textArea) {
+            area =  textArea;
```
Comment (turbanoff) [inline test/jdk/performance/client/SwingMark/src/AdvancedTextAreaTest.java:188]:
```suggestion
            area = textArea;
```

File: test/jdk/performance/client/SwingMark/src/AdvancedTextAreaTest.java [hunk 14]
```diff
@@ -0,0 +1,244 @@
...
+    }
+
+    static class TextSelector implements Runnable {
+        JTextArea area;
+        int start;
+        int end;
+        Random random;
+
+        public TextSelector(JTextArea textArea) {
+            area =  textArea;
```
Comment (turbanoff) [inline test/jdk/performance/client/SwingMark/src/AdvancedTextAreaTest.java:203]:
```suggestion
            area = textArea;
```

File: test/jdk/performance/client/SwingMark/src/TableColMoveTest.java [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+      runStandAloneTest(new TableColMoveTest());
+   }
+}
+
+
+class TableColMover implements Runnable {
+    JTable table;
+
+    public TableColMover(JTable table) {
+        this.table =  table;
```
Comment (turbanoff) [inline test/jdk/performance/client/SwingMark/src/TableColMoveTest.java:94]:
```suggestion
        this.table = table;
```

File: test/jdk/performance/client/SwingMark/src/TextPaneTest.java [hunk 16]
```diff
@@ -0,0 +1,114 @@
...
+            paintCount++;
+        }
+    }
+
+    static class TextAppender implements Runnable {
+        JTextPane area;
+        String appendString;
+
+        public TextAppender(JTextPane textArea, String appendThis) {
+            area =  textArea;
```
Comment (turbanoff) [inline test/jdk/performance/client/SwingMark/src/TextPaneTest.java:102]:
```suggestion
            area = textArea;
```

**Review by avu — APPROVED** [review]
Looks good

**Pull request closed without merging**